blueprints/data-solutions/vertex-mlops/main.tf

```diff
@@ -0,0 +1,230 @@
+        for member in members : { role = role, member = member }
+      ]
+    ]) : "${binding.role}-${binding.member}" => binding
+  }
+}
+
+module "gcs-bucket" {
+  count          = var.bucket_name == null ? 0 : 1
+  source         = "../../../modules/gcs"
+  project_id     = module.project.project_id
+  name           = var.bucket_name
+  prefix         = var.prefix
+  location       = var.region
+  storage_class  = "REGIONAL"
+  versioning     = false
+  encryption_key = try(local.service_encryption_keys.storage, null) # Example assignment of an encryption key
+}
+
+# Default bucket for Cloud Build to prevent error: "'us' violates constraint ‘constraints/gcp.resourceLocations’"
+# https://stackoverflow.com/questions/53206667/cloud-build-fails-with-resource-location-constraint
+module "gcs-bucket-cloudbuild" {
+  source         = "../../../modules/gcs"
+  project_id     = module.project.project_id
+  name           = "${var.project_id}_cloudbuild"
+  prefix         = var.prefix
+  location       = var.region
+  storage_class  = "REGIONAL"
+  versioning     = false
+  encryption_key = try(local.service_encryption_keys.storage, null) # Example assignment of an encryption key
+}
+
+module "bq-dataset" {
+  count          = var.dataset_name == null ? 0 : 1
+  source         = "../../../modules/bigquery-dataset"
+  project_id     = module.project.project_id
+  id             = var.dataset_name
+  location       = var.region
+  encryption_key = try(local.service_encryption_keys.bq, null) # Example assignment of an encryption key
+}
+
+module "vpc-local" {
+  count      = local.use_shared_vpc ? 0 : 1
+  source     = "../../../modules/net-vpc"
+  project_id = module.project.project_id
+  name       = "default"
+  subnets = [
+    {
+      "name" : "default",
+      "region" : "${var.region}",
+      "ip_cidr_range" : "10.4.0.0/24",
+      "secondary_ip_range" : null
+    }
+  ]
+  psa_config = {
+    ranges = {
+      "vertex" : "10.13.0.0/18"
+    }
+    routes = null
+  }
+}
+
+module "firewall" {
+  count      = local.use_shared_vpc ? 0 : 1
+  source     = "../../../modules/net-vpc-firewall"
+  project_id = module.project.project_id
+  network    = module.vpc-local[0].name
+  default_rules_config = {
+    disabled = true
+  }
+  ingress_rules = {
+    dataflow-ingress = {
+      description          = "Dataflow service."
+      direction            = "INGRESS"
+      action               = "allow"
+      sources              = ["dataflow"]
+      targets              = ["dataflow"]
+      ranges               = []
+      use_service_accounts = false
+      rules                = [{ protocol = "tcp", ports = ["12345-12346"] }]
+      extra_attributes     = {}
+    }
+  }
+
+}
+
+module "cloudnat" {
+  count          = local.use_shared_vpc ? 0 : 1
+  source         = "../../../modules/net-cloudnat"
+  project_id     = module.project.project_id
+  region         = var.region
+  name           = "default"
```

Comment:
I would not use default as name of the cloudnat, I would at at least the prefix

Reply:
Added in the TODO list